opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/ReadOnlyApi.java

```diff
@@ -0,0 +1,67 @@
+package de.geeksfactory.opacclient.apis;
+
+import org.json.JSONException;
+
+import java.io.IOException;
+
+import de.geeksfactory.opacclient.objects.Account;
+import de.geeksfactory.opacclient.objects.AccountData;
+import de.geeksfactory.opacclient.objects.DetailledItem;
+import de.geeksfactory.opacclient.objects.Library;
+
+/**
+ * For "read-only" API implementations, i.e. without account support, some abstract method
+ * implementations are provided.
+ */
+public abstract class ReadOnlyApi extends BaseApi {
```

Comment:
Yes, nice idea. I am not 100% satisfied with the name, however: In the literal sense, a `ReadOnlyApi` would for example also provide the `account` function, because it only *reads* data from the account. Maybe `SearchOnlyApi` or `NoAccountApi` would be better?